components/ds3231/ds3231.c

```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
     return ((val / 10) << 4) + (val % 10);
 }
 
+#include <stdio.h>
+
+// Function to check if a year is a leap year
+int isLeapYear(int year) { return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0); }
```

Comment:
1. No CamelCase allowed.
2. No one-line functions format allowed
3. This is static function
4. This is inline function